services/connector/database/persistence.go

```diff
@@ -6,29 +6,32 @@ import (
 	"github.com/status-im/status-go/crypto/types"
 )
 
-const upsertDAppQuery = "INSERT INTO connector_dapps (url, name, icon_url, shared_account, chain_id) VALUES (?, ?, ?, ?, ?) ON CONFLICT(url) DO UPDATE SET name = excluded.name, icon_url = excluded.icon_url, shared_account = excluded.shared_account, chain_id = excluded.chain_id"
-const selectDAppByUrlQuery = "SELECT name, icon_url, shared_account, chain_id FROM connector_dapps WHERE url = ?"
-const selectDAppsQuery = "SELECT url, name, icon_url, shared_account, chain_id FROM connector_dapps"
-const deleteDAppQuery = "DELETE FROM connector_dapps WHERE url = ?"
+const upsertDAppQuery = "INSERT INTO connector_dapps (url, name, icon_url, client_id, shared_account, chain_id) VALUES (?, ?, ?, ?, ?, ?) ON CONFLICT(url, client_id) DO UPDATE SET name = excluded.name, icon_url = excluded.icon_url, shared_account = excluded.shared_account, chain_id = excluded.chain_id"
+const selectDAppByUrlAndClientIDQuery = "SELECT name, icon_url, shared_account, chain_id FROM connector_dapps WHERE url = ? AND client_id = ?"
+const selectDAppsQuery = "SELECT url, name, icon_url, client_id, shared_account, chain_id FROM connector_dapps"
+const deleteDAppQuery = "DELETE FROM connector_dapps WHERE url = ? AND client_id = ?"
 
 type DApp struct {
 	URL           string        `json:"url"`
 	Name          string        `json:"name"`
 	IconURL       string        `json:"iconUrl"`
+	ClientID      string        `json:"clientId"`
 	SharedAccount types.Address `json:"sharedAccount"`
 	ChainID       uint64        `json:"chainId"`
 }
 
 func UpsertDApp(db *sql.DB, dApp *DApp) error {
-	_, err := db.Exec(upsertDAppQuery, dApp.URL, dApp.Name, dApp.IconURL, dApp.SharedAccount, dApp.ChainID)
+	_, err := db.Exec(upsertDAppQuery, dApp.URL, dApp.Name, dApp.IconURL, dApp.ClientID, dApp.SharedAccount, dApp.ChainID)
 	return err
 }
 
-func SelectDAppByUrl(db *sql.DB, url string) (*DApp, error) {
+func SelectDAppByUrlAndClientID(db *sql.DB, url string, clientID string) (*DApp, error) {
```

Comment:
`AND` into the function name is a good marker of that something is going wrong 😄 
The arguments are obvious from the arguments list really, let's just drop it:
```suggestion
func SelectDApp(db *sql.DB, url string, clientID string) (*DApp, error) {
```

Reply:
fixed